inaturalist/INaturalistIOS
# ExploreCommentCell: Support hyperlinks in observation comments

UITextView has built-in support for hyperlinks, so use a non-editable UITextVIew in place of a UILabel.
This is one possible approach that can be taken to support hyperlinks in observation comments, as requested in https://github.com/inaturalist/INaturalistIOS/issues/218. This only made the change in ExploreCommentCell, but the same idea could be easily ported to the other views mentioned in that issue.

Note: I used some UITextView properties only available on iOS 7+ to remove the unwanted default top padding on UITextView, if the project supports older versions of iOS, I can use another approach that isn't as elegant.

Test Plan:

1. Navigate to an observation with a comment that has a hyperlink (search for "Western Subterranean Termite" near San Francisco, and select @loarie's observation or just add one yourself)
2. Ensure that the hyperlink is clickable and opens in Safari (if we wanted to, we could spend some time and implement opening certain hyperlinks directly in-app, but this would be a lot more involved).

Tested on iPhone 5S, 6, and iPad Air Simulators

![simulator screen shot dec 10 2015 10 24 09 pm](https://cloud.githubusercontent.com/assets/821279/11738178/361640c0-9f95-11e5-84d7-6e98bb25cac8.png)
